Cargo.toml

```diff
@@ -1,14 +1,7 @@
-[package]
-name = "tracing_sim"
-version = "0.1.0"
-authors = ["Anirudh Sivaraman <sk.anirudh@gmail.com>", "Jessica Berg <jb7399@nyu.edu>", "Fabian Ruffy <fruffy@nyu.edu>"]
-edition = "2018"
+[workspace]
```

Comment:
Split the toml into three separate workspaces. Now all the libs will be automatically built when running `cargo build`.

Reply:
Added a minimal README. We can improve this later. I am not sure how the compiler workflow is currently.